opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -225,7 +225,7 @@ def test_span_processor_for_source(self):
     def test_start_span_implicit(self):
```

Comment:
can you add a unit test validating that #881 is fixed? 

Adding a unit test, with a clear explanation on why this behavior is important, would help someone who wants to refactor in the future and regress this behavior.